xinference/model/embedding/core.py

```diff
@@ -595,6 +595,65 @@ def encode(
 
             return all_embeddings, all_token_nums
 
+        @no_type_check
+        def _encode_clip(
+            model: SentenceTransformer,
+            sentences: Union[str, List[str]],
+            convert_to_numpy: bool = True,
+            **kwargs,
+        ):
+            import base64
+            import json
+            import re
+            from io import BytesIO
+
+            from PIL import Image
+
+            def base64_to_image(base64_str: str) -> Image.Image:
+                # base64_data = re.sub("^data:image/.+;base64,", "", base64_str)
+                base64_data = base64_str.split(",", 1)[1]
+                byte_data = base64.b64decode(base64_data)
+                image_data = BytesIO(byte_data)
+                img = Image.open(image_data)
+                return img
+
+            all_token_nums = 0
+            all_embeddings = []
+            texts = []
+            images = []
+
+            for data in sentences:
+                data = json.loads(data.replace("'", '"'))
```

Comment:
https://github.com/xorbitsai/inference/blob/ac9ddca75345780533858aa6cf88397c802f571c/xinference/api/restful_api.py#L97

Now the embedding API only allow list and similar type which follows the openai API, but now we can extend it to support dict, to be specific, Dict[str, str] to be passed for multi-model embedding.

IMO we should directly support dict type instead of parsing json here.